src/NHibernate/Cfg/MappingSchema/HbmElement.cs

```diff
@@ -71,5 +71,24 @@ public HbmType Type
 		}
 
 		#endregion
+
+		/// <summary>
+		/// Columns and Formulas, in declared order
+		/// </summary>
+		[XmlIgnore]
+		public IEnumerable<object> ColumnsAndFormulas
+		{
+			get
+			{
+				if (Items != null)
+					return Items;
+				// Avoid a possible breaking change for mapping having left a column attribute along with a formula one.
```

Comment:
Is this logic deliberate? In `HbmManyToOne` currently is 
```csharp
if (Items != null)
	return Items;

return AsColumns()
	.Concat(AsFormulas().Cast<object>());
```
(if inline `Columns` and `Formulas` fields)

Reply:
Yes, and this was a possible breaking change at the time it was coded. Since it was not hard to avoid it and keep the previous behavior for `element` and other cases, I have done it that way.
This means keeping in place an inconsistency in behavior between `many-to-one` and others.

If instead we accept doing a possible breaking change here (for somewhat erroneous mappings anyway: it is not documented what happens when specifying both attributes; in my mind they ideally should have been mutually exclusive; people wanting to declare a column and a formula should use tags instead of attributes), we have three possibilities:

 - Align them all on the `many-to-one` behavior. This behavior causes a tag declaring both a `column` attribute and a `formula` attribute to be considered as spanning two columns, the first being a regular table column and the second being a formula. Why not, but this looks a bit arbitrary, nothing intuitive in knowing which one would be considered as the first or second.
- Align them all on this PR `element` behavior. This behavior causes a tag declaring both a `column` attribute and a `formula` attribute to be considered as declaring only a formula, ignoring the column. Why not, but likewise this looks a bit arbitrary, nothing intuitive in knowing which one would be taken or ignored.
- "Break" them all by throwing an explicit `MappingException` when both attributes are specified. To be consistent, we should then throw also if one of them is specified along with sub-tags. (Currently this case ignores the attributes.)

Reply:
Probably mapping exception is preferred option.